src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs

```diff
@@ -0,0 +1,210 @@
+                if (found < 0) return false;
+
+                if (found + 3 < length)
+                {
+                    if (buffer[found + 1] == (byte)'\n' &&
+                        buffer[found + 2] == (byte)'\r' &&
+                        buffer[found + 3] == (byte)'\n')
+                    {
+                        return true;
+                    }
+                }
+
+                offset = found + 1;
+            }
+        }
+        private Uri CreateServer(string response)
+        {
+            Socket listen = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            listen.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+            listen.Listen(1);
+
+            listen.AcceptAsync().ContinueWith(async t =>
+            {
+                Socket accept = t.Result;
+                accept.NoDelay = true;
+                Stream s = GetStream(new NetworkStream(accept));
+
+                // Read whatever request the client sent us
+                // Note, we assume there's no content
+                byte[] buffer = new byte[4096];
+                int length = 0;
+                while (true)
+                {
+                    int bytesRead = await s.ReadAsync(buffer, length, buffer.Length - length);
+                    if (bytesRead == 0)
+                        throw new InvalidOperationException($"unexpected end of stream, buffer={Encoding.UTF8.GetString(buffer, 0, length)}");
+
+                    length += bytesRead;
+
+                    if (GetHeaders(buffer, length))
+                        break;
+                }
+
+                // Send the specified response
+                byte[] responseBytes = Encoding.UTF8.GetBytes(response);
+                await s.WriteAsync(responseBytes, 0, responseBytes.Length);
+                accept.Close();
+                listen.Close();
+            }, TaskContinuationOptions.ExecuteSynchronously);
+
+            IPEndPoint ep = (IPEndPoint)listen.LocalEndPoint;
+            return new Uri($"http://{ep.Address}:{ep.Port}/");
+        }
+
+        [Theory]
+        [InlineData(200, "OK")]
+        [InlineData(200, "Sure why not?")]
+        [InlineData(200, "")]
+        [InlineData(200, " ")]
+        [InlineData(200, "      ")]
+        [InlineData(200, "      Something")]
+        [InlineData(200, "\t")]
+        [InlineData(200, "O K")]
+        [InlineData(200, "O\tK")]
+        [InlineData(200, "O    \t\t  \t\t\t\t  \t K")]
+        [InlineData(201, "Created")]
+        [InlineData(202, "Accepted")]
+        [InlineData(299, "This is not a real status code")]
+        [InlineData(345, "redirect to nowhere")]
+        [InlineData(400, "Bad Request")]
+        [InlineData(500, "Internal Server Error")]
+        [InlineData(555, "we just don't like you")]
+        [InlineData(600, "still valid")]
+        [InlineData(999, "this\ttoo\t")]
+        public async Task ValidStatusLine(int statusCode, string reason)
+        {
```

Comment:
Please follow the current naming convention for networking tests which use a three part name separated by underscores.